### PR TITLE
Add @warn_unused_result annotations

### PR DESCRIPTION
**Pull request by neilpa**
Note that this PR targets #2099

### CODE REVIEW

File: ReactiveCocoa/Swift/Action.swift [hunk 1]
```diff
@@ -86,6 +86,7 @@ public final class Action<Input, Output, Error: ErrorType> {
 	/// If the action is disabled when the returned SignalProducer is started,
 	/// the produced signal will send `ActionError.NotEnabled`, and nothing will
 	/// be sent upon `values` or `errors` for that particular signal.
+	@warn_unused_result(message="Did you forget to `start` the producer?")
```
Comment (JaviSoto) [inline ReactiveCocoa/Swift/Action.swift]: Woot this is really cool!!!!

**Comment by JaviSoto** [comment]
This is awesome. I wish there was a way to factor this out somehow, there's a lot of "duplication", but it's not really code per-se so it's not a huge deal.

:shipit: 

(I'll merge this already because it targets a different PR. If anybody else disagrees with this we can always revert it)

**Comment by neilpa** [comment]
:sparkles:

**Comment by neilpa** [comment]
> (I'll merge this already because it targets a different PR. If anybody else disagrees with this we can always revert it)

Figured it would be nice to have a separate history for this which is I why I didn't include it directly in the protocol extension branch.

**Comment by neilpa** [comment]
Oh, and here's what it actually looks like

screen shot 2015-06-18 at 2 24 40 pm

**Comment by nomothetis** [comment]
If I may, the "`" looks a bit like a quote, which reads wrong. I'd say change it either "Did you forget to call `start` on the producer?" or simply "Did you forget to start the producer?"

**Comment by neilpa** [comment]
Done 30ba8ad

**Comment by nomothetis** [comment]
:bow:

**Comment by NachoSoto** [comment]
🎊

File: ReactiveCocoa/Swift/Action.swift (hunk 1 above)
Comment (MarcoSero) [inline ReactiveCocoa/Swift/Action.swift]: :100:

**Comment by jspahrsummers** [comment]
:tada:

**Pull request merged**